### PR TITLE
feat: Top search component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

Adds `Top Search` component, an element that will be part of the `Suggestion` [dropdown component](https://vtex-dev.atlassian.net/browse/FSSS-202).
This component attempts to list the ranking of the top searched items. Since this task  doesn't consider the integration part yet, this component is receiving an array of `searchedItems: LinkItem[]`. It needs to be revised later during the integration.

```
type LinkItem = {
  href: string       //link to redirect when the item is clicked
  name: string    //label to display on the list
}
```

## How does it work?

https://user-images.githubusercontent.com/3356699/155030110-b89b946e-b14f-45b5-bfab-0dcb19eaf930.mov

## How to test it?
There is no preview for it yet. (Because it depends on the Suggestion component)
You can call this component on any page to test it visually.

e.g.:

- Example of `searchedItems`
```
const searchedItems = [
  { name: 'Office Supplies', href: '/' },
  { name: 'Headphones', href: '/' },
  { name: 'Notebooks', href: '/' },
]
```
 
- Calling the component:
`<SuggestionsTopSearch searchedItems={searchedItems} />`

## References

[FSSS-194 - Top Search Component](https://vtex-dev.atlassian.net/jira/software/projects/FSSS/boards/632?selectedIssue=FSSS-194)

## Checklist

<em>You may erase this after checking them all ;)</em>

- [x] CHANGELOG entry added

### CODE REVIEW

**Review by eduardoformiga — APPROVED** [review]
Good Job! ✨ 

My mind is in an infinite loop wondering which name is better: `SuggestionTopSearch` or `TopSearchSuggestion`. Thus, this hard task is up to you hahaha

**Comment by hellofanny** [comment]
> Good Job! ✨
> 
> My mind is in an infinite loop wondering which name is better: `SuggestionTopSearch` or `TopSearchSuggestion`. Thus, this hard task is up to you hahaha

hahahah I had the same trouble. but since it's related to `Suggestions` and inside this folder I decided to follow `Footer` pattern 🙈

**Comment by ArthurTriis1** [comment]
Good! ✨ 
I lIke SuggestionTopSearch too, is easier to group the Suggestions parts.

**Review by ArthurTriis1 — APPROVED** [review]

File: src/components/search/Suggestions/SuggestionsTopSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  SuggestionsTopSearchProps
+>(function SuggestionsTopSearch(
+  { testId = 'top-search', searchedItems, ...otherProps },
+  ref
+) {
+  return (
+    <section
+      ref={ref}
+      data-testid={testId}
+      data-suggestions-top-search
```
Comment (lucasfp13) [inline src/components/search/Suggestions/SuggestionsTopSearch.tsx:38]:
```suggestion
      data-store-suggestions-top-search
```

**Review by lucasfp13 — APPROVED** [review]
LGTM!

File: src/components/search/Suggestions/suggestions.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  font-size: var(--text-size-3);
+  line-height: 1.5;
+}
+
+.suggestions__section {
+  padding: var(--space-1) var(--space-3) var(--space-2);
+  background-color: var(--bg-neutral-lightest);
+}
+
+[data-suggestions-top-search] {
```
Comment (hellofanny) [inline src/components/search/Suggestions/suggestions.scss:13]:
```suggestion
[data-store-suggestions-top-search] {
```

**Pull request merged**